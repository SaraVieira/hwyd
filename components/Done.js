/* @flow */

import React, { Component } from 'react';
import { View } from 'react-native';
import styled from 'styled-components';
import { DangerZone } from 'expo';

import doneJSON from '../assets/animations/done.json';

type State = {
  width: number,
  height: number,
};

type LayoutEvent = {
  nativeEvent: {
    layout: {
      width: number,
      height: number,
    },
  },
};

let { Lottie } = DangerZone;

const Container = styled(View)`
  background-color: transparent;
  align-items: center;
  justify-content: center;
`;

export default class Done extends Component<*, State> {
  animation: Lottie;

  state = {
    width: 0,
    height: 0,
  };

  componentDidUpdate = () => {
    if (this.state.width > 0 && this.state.height > 0) {
      this.animation.reset();
      this.animation.play();
    }
  };

  render = () => {
    return (
      <Container onLayout={this.onLayout} style={this.props.style}>
        {this.state.width > 0 &&
          this.state.height > 0 && (
            <Lottie
              loop={false}
              style={{
                height: this.state.height,
                width: this.state.width,
              }}
              ref={ref => {
                this.animation = ref;
              }}
              source={doneJSON}
            />
          )}
      </Container>
    );
  };

  onLayout = ({ nativeEvent: { layout: { width, height } } }: LayoutEvent) => {
    if (width === 0 || height === 0) return;
    this.setState(() => ({ width, height }));
  };
}
