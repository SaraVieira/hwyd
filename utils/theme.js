import {
  /* colors */
  APPLE_VALLEY,
  BREWED_MUSTARD,
  CREAMY_PEACH,
  OLD_GERANIUM,
  PENCIL_LEAD,
  ROGUE_PINK,
  ROSY_HIGHLIGHT,
  SOFT_BLUE,
  SQUEAKY,
  WHITE,

  /* fonts */
  BOLD_FONT,
  ITALIC_FONT,
  REGULAR_FONT,
  SUBTITLE_FONT,
  TITLE_FONT,
} from './constants';

export default {
  colors: {
    appleValley: APPLE_VALLEY,
    brewedMustard: BREWED_MUSTARD,
    creamyPeach: CREAMY_PEACH,
    oldGeranium: OLD_GERANIUM,
    pencilLead: PENCIL_LEAD,
    roguePink: ROGUE_PINK,
    rosyHighlight: ROSY_HIGHLIGHT,
    softBlue: SOFT_BLUE,
    squeaky: SQUEAKY,
    white: WHITE,
  },
  fonts: {
    bold: BOLD_FONT,
    italic: ITALIC_FONT,
    regular: REGULAR_FONT,
    subtitle: SUBTITLE_FONT,
    title: TITLE_FONT,
  },
};
