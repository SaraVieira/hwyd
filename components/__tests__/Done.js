import React from 'react';
import { ThemeProvider } from 'styled-components';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Done from '../Done';
import theme from '../../utils/theme';

describe('<Done />', () => {
  it('renders a done with check in animation', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <Done />
      </ThemeProvider>,
    );
    const root = tree.find(Done).shallow();
    expect(root).toMatchSnapshot();
  });
});
