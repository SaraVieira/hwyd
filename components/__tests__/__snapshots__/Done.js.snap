// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Done /> renders a done with check in animation 1`] = `
<Styled(View)
  onLayout={[Function]}
/>
`;
